test/integration/files.js

```diff
@@ -59,6 +60,47 @@ describe('files API', async function () {
     created.attributes.md5sum.should.equal('7Zfd8PaeeXsm5WJesf/KJw==')
   })
 
+  it('fails if the contentLength is too small', async function () {
+    const filename = 'foo_' + random()
+    const stream = new Readable()
+
+    stream.push('datastring1')
+    stream.push(null)
+
+    const created = await cozy.client.files.create(stream, {
+      name: filename,
+      contentType: 'application/json',
+      contentLength: 4
+    })
+
+    // created.should.be.rejectedWith(/content size/)
+    created.should.have.property('attributes')
+    created.attributes.md5sum.should.equal('jXd/OF09/siBXSD3SWAm3A==') // md5 "data"
+  })
+
+  it('fails if the contentLength is too big', async function () {
+    const filename = 'foo_' + random()
+    const stream = new Readable()
+    this.timeout(5000)
+
+    stream.push('datastring1')
+    stream.push(null)
+    // stream.timeout = 2000
+
+    const oldFetch = global.fetch
+    global.fetch = (url, opts) => {
+      if (opts) opts.timeout = 500
+      return oldFetch(url, opts)
+    }
+    after(() => { global.fetch = oldFetch })
```

Comment:
I did not know about this after() function and I cannot see it in the [Jest doc](https://facebook.github.io/jest/docs/en/api.html). Where does it come from ?

Reply:
OK, we use mocha here.